### PR TITLE
Refactor validation tests to use dynamic imports and add edge case tests

### PR DESCRIPTION
**Pull request by triche**
- Removed the CommonJS setup for TypeScript in validation tests.
- Converted validation tests to use async/await with dynamic imports.
- Added new edge case tests for diagram and palette validation.
- Consolidated test structure for better readability and maintainability.
- Deleted the old ES module validation test file.
- Introduced a new test file for transformation apply and export tests, covering various scenarios including internal wiring and duplicate port names.

### CODE REVIEW

**Comment by triche** [comment]
@copilot fix tests

File: src/App.tsx [hunk 1]
```diff
@@ -76,9 +81,48 @@ function App() {
...
+    const snapshot = stack.pop()
+    if (!snapshot) return
+    setNodes(snapshot.nodes)
+    setWires(snapshot.wires)
+    setSelectedNodeIds([])
+    setSelectedWireIds([])
+    setContextMenu(null)
+    setWireDraft(null)
+  }
+
```
[the remark this answers is not in the document]
Comment (triche) [reply src/App.tsx:125]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/triche/DxT/pull/30#discussion_r2724767441)

File: test/transformation-applicability.test.cjs [hunk 2]
```diff
@@ -100,28 +95,37 @@ const transformations = [
...
+        const selection = ['lonely-1'];
+        const applicable = getApplicableTransformations(nodes, wires, selection, transformations);
+        assertTrue(applicable.some(t => t.name === 'Filter + Normalize -> OptimizedFilter'), 'Unconnected ports should still match');
+    });
+
+    console.log(`\n=== Test Summary: ${passed}/${total} passed ===`);
+    if (passed !== total) {
+        process.exitCode = 1;
+    }
+};
```
[the remark this answers is not in the document]
Comment (triche) [reply test/transformation-applicability.test.cjs:148]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/triche/DxT/pull/30#discussion_r2724767444)
